config.py

```diff
@@ -46,3 +46,34 @@
     "tag_failure": False,
     "tag_outcome": False,
 }
+
+# JSONschema schema used to validate config files
+CONFIG_FILE_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "name": {"type": "string"},
+        "version": {"type": "string"},
+        "target": {"type": "string"},
+        "features": {"type": "array"},
+        "batch_id": {"type": "string"},
+        "train_on": {
+            "type": "object",
+            "properties": {
+                "start_date": {"type": "string"},
+                "end_date": {"type": "string"},
+                "sample_size": {"type": "integer"},
+            },
+            "required": ["start_date", "end_date", "sample_size"],
+        },
+        "predict_on": {"type": "string"},
```

Comment:
suggestion: peut être ajouter une prop pour indiquer qu'on s'attend à avoir une date, ici ?

```suggestion
        "predict_on": {"type": "string", "format": "date-time"},
```

...et idem pour les autres props qui sont des dates.